### PR TITLE
chore: upgrade @anthropic-ai/sdk to 0.104.1 and @anthropic-ai/vertex-sdk to 0.17.1

### LINKED ISSUE #438: [BUG] HTTP proxy is ignored on Anthropic provider
### Problem (one or two sentences)

Currently VSCode HTTP proxy configuration is being ignored when using Anthropic.

### Context (who is affected and when)

Any user for who internet access is available only through a proxy.

### Reproduction steps

In my particular case I'm doing remote development on a container from which internet access is only available through a HTTP proxy. HTTP proxy is setup as http://proxy:1080 where `proxy` is the sidecar container providing proxy functionality. All attempts to use Anthropic provider fail. 

### Expected result

Anthropic provider does not fail

### Actual result

I get Provider Error with "Connection error." as error details

### Variations tried (optional)

_No response_

### App Version

Roo code 3.50.4

### API Provider (optional)

Anthropic

### Model Used (optional)

_No response_

### Relevant logs or errors (optional)

```shell

```

### PR DESCRIPTION
**Pull request by p12tic**
### Related GitHub Issue

Closes: #438

### Description

Upgrades Anthropic AI SDK to newer version.

### Test Procedure

Check if Anthropic provider still runs fine.

### Pre-Submission Checklist

- [x] **Issue Linked**: This PR is linked to an approved GitHub Issue (see "Related GitHub Issue" above).
- [x] **Scope**: My changes are focused on the linked issue (one major feature/fix per PR).
- [x] **Self-Review**: I have performed a thorough self-review of my code.
- [x] **Testing**: New and/or updated tests have been added to cover my changes (if applicable).
- [x] **Documentation Impact**: I have considered if my changes require documentation updates (see "Documentation Updates" section below).
- [x] **Contribution Guidelines**: I have read and agree to the [Contributor Guidelines](/CONTRIBUTING.md).

### Screenshots / Videos

N/A.

### Documentation Updates

Not needed.

### Additional Notes

None.

### Get in Touch

Mention at github.

## Summary by CodeRabbit

* **Bug Fixes**
  * Fixed image handling to respect image source type across providers/transformers: base64 images are converted to data URLs, URL images use their URL, and unsupported types fall back to an appropriate placeholder.
  * Corrected tool-result content mapping so non-text items don’t incorrectly contribute text (with consistent placeholders where images are involved).
* **New Features**
  * Extended markdown export to recognize and render additional block types (document, search result, tool reference), including reasoning-related blocks.
* **Tests / Chores**
  * Expanded unit and e2e coverage for image/tool-result edge cases; updated SDK dependency versions and test runner commands.

### CODE REVIEW

File: src/api/transform/openai-format.ts [hunk 1]
```diff
@@ -361,7 +361,10 @@ export function convertToOpenAiMessages(
 										toolResultImages.push(part)
```
Comment (edelauna) [inline src/api/transform/openai-format.ts]: Should this push be guarded with `if (part.source.type === "base64")`? A URL-sourced image would get pushed here and emit `"(see following user message for image)"`, but the follow-up flush block at line 385 is commented out — so the image silently disappears while the placeholder still goes to the model.

File: src/api/transform/openai-format.ts [hunk 2]
```diff
@@ -361,7 +361,10 @@ export function convertToOpenAiMessages(
...
 										return "(see following user message for image)"
 									}
-									return part.text
+									if (part.type === "text") {
+										return part.text
+									}
+									return ""
```
Comment (edelauna) [inline src/api/transform/openai-format.ts:370]: In SDK 0.104.1, `ToolResultBlockParam.content` is `Array<TextBlockParam | ImageBlockParam | SearchResultBlockParam | DocumentBlockParam | ToolReferenceBlockParam>`. Could a `SearchResultBlockParam` or `DocumentBlockParam` inside a tool result end up here and get silently dropped as `""`?

File: src/api/transform/openai-format.ts [hunk 3]
```diff
@@ -422,12 +425,18 @@ export function convertToOpenAiMessages(
...
-									return {
-										type: "image_url",
-										image_url: { url: `data:${part.source.media_type};base64,${part.source.data}` },
+									if (part.source.type === "base64") {
+										return {
+											type: "image_url",
+											image_url: { url: `data:${part.source.media_type};base64,${part.source.data}` },
+										}
 									}
+									return { type: "text", text: "[Image]" }
```
Comment (edelauna) [inline src/api/transform/openai-format.ts:445]: OpenAI chat completions supports URL images natively via `{ type: "image_url", image_url: { url: "https://..." } }`. `ai-sdk.ts` already handles this case at line 63 as a reference. Worth forwarding URL sources here instead of falling back to `[Image]`?

File: src/api/transform/mistral-format.ts [hunk 4]
```diff
@@ -104,14 +104,20 @@ export function convertToMistralMessages(anthropicMessages: Anthropic.Messages.M
...
-									},
+								if (part.source.type === "base64") {
+									return {
+										type: "image_url",
+										imageUrl: {
+											url: `data:${part.source.media_type};base64,${part.source.data}`,
+										},
+									}
 								}
+								return { type: "text", text: "[Image]" }
```
Comment (edelauna) [inline src/api/transform/mistral-format.ts:121]: Mistral also supports URL images natively (`imageUrl: { url: "https://..." }`). Could forward `source.type === "url"` here instead of substituting `[Image]`, similar to `openai-format.ts`.

File: src/api/transform/vscode-lm-format.ts [hunk 5]
```diff
@@ -72,11 +72,19 @@ export function convertToVsCodeLmMessages(
 								? [new vscode.LanguageModelTextPart(toolMessage.content)]
 								: (toolMessage.content?.map((part) => {
 										if (part.type === "image") {
+											if (part.source.type === "base64") {
+												return new vscode.LanguageModelTextPart(
+													`[Image (${part.source?.type || "Unknown source-type"}): ${part.source?.media_type || "unknown media-type"} not supported by VSCode LM API]`,
```
Comment (edelauna) [inline src/api/transform/vscode-lm-format.ts]: Inside `if (part.source.type === "base64")`, `part.source?.type` is always `"base64"` — the optional chain and `|| "Unknown source-type"` fallback are dead. Could simplify to `[Image (base64): ${part.source.media_type} not supported by VSCode LM API]`.

**Review by edelauna — CHANGES_REQUESTED** [review]
Thank you for taking on this dependency bump. Had a couple comments regarding image handling.

**Review by edelauna — DISMISSED** [review]

**Review by edelauna — APPROVED** [review]

**Comment by p12tic** [comment]
@edelauna Thanks for taking over and sorry for not finishing the PR in time.
